bin/common.ml

```diff
@@ -763,6 +764,33 @@ module Builder = struct
                ])
             Automatic
         & info [ "file-watcher" ] ~doc)
+    and+ watch_exclusions =
+      let doc_escape s =
+        let escape_char = function
+          | '$' -> "\\$"
+          | c -> Char.escaped c
+        in
+        String.to_seq s |> List.of_seq |> List.map ~f:escape_char
+        |> String.concat ~sep:""
+      in
+      let escaped_default_exclusions =
+        List.map ~f:doc_escape Dune_rules.Workspace.default_watch_exclusions
+      in
+      let doc =
+        "Adds a regular expression that will exclude matching directories from \
```

Comment:
I guess we have to settle on a choice of regexp syntax: `str`, PCRE, `emacs`, POSIX, etc. Also, if I understand correctly, every regexp passed using this flag is _added_ to the standard set. I think it would be a good idea to be explicit about this. (But see below.)

Reply:
> I guess we have to settle on a choice of regexp syntax: `str`, PCRE, `emacs`, POSIX, etc.

It was already POSIX in dune_file_watcher.ml, so I won't change that (but will be explicit in documentation).

> Also, if I understand correctly, every regexp passed using this flag is _added_ to the standard set. I think it would be a good idea to be explicit about this. 

Any use of the `--watch-exclusions` replaces the standard exclusions. I've switched the documentation to be super-explicit.